recipes/newrelic/infrastructure/super-agent/debian.yml

```diff
@@ -422,9 +448,98 @@ install:
           else
             sed -i 's/\(endpoint: https:\/\/opamp.\).*/\1'"service.newrelic.com\/v1\/opamp"'/' /etc/newrelic-super-agent/config.yaml
           fi
+
+    config_super_agent_auth:
+      cmds:
         - |
-          if [ "{{.NEW_RELIC_SUPER_AGENT_FLEET_ENABLED}}" != "false" ] ; then
-            sed -i 's/api-key: API_KEY_HERE/api-key: {{.NEW_RELIC_LICENSE_KEY}}/g' /etc/newrelic-super-agent/config.yaml
+          if [ "{{.NEW_RELIC_SUPER_AGENT_FLEET_ENABLED}}" = "false" ] && [ "{{ .NEW_RELIC_ORGANIZATION }}" != "" ]; then
```

Comment:
NEW_RELIC_ORGANIZATION is required to create the identity but why do we need the && in here? Shouldn't this only check for NEW_RELIC_SUPER_AGENT_FLEET_ENABLED = false and always enter?

Reply:
Good point. `NEW_RELIC_ORGANIZATION` is mandatory now for the creation of the identity but it might not be in the future.

Done in 200148ea3a8b09fd5f0a71a0ab51f3cb760fa1e5